### ISSUE TITLE
Issue with aligning aliasses when using jinja

### ISSUE THREAD

**Issue by ferdyh** [sqlfluff/sqlfluff#5429]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When I have a simple query containing jinja, i get the fault that LT01 is voilated, alias_expressions are expected to be aligned.

```sql
select
    {{ generate_surrogate_key('test', ['a', 'b', 'c']) }} as test_key,
    b as b_col
from {{ ref("test") }}
```

When I align the aliasses like below, I still get the same error

```sql
select
    {{ generate_surrogate_key('test', ['a', 'b', 'c']) }} as test_key,
    b                                                     as b_col
from {{ ref("test") }}
```

When I use sqlfluff to 'fix' this issue, I get the following result, but this looks like its aligning on the macro's result (or something else). The following violation is LT05, line too long.

```sql
select
    {{ generate_surrogate_key('test', ['a', 'b', 'c']) }} as test_key,
    b                                                                                                                                        as b_col
from {{ ref("test") }}
```

### Expected Behaviour

SQLFluff to accept alignment on the actual jinja, not on any parsed result.

### Observed Behaviour

SQLFluff seems to align on a generated macro, at least not whats visible in the editor

### How to reproduce

Use the supplied queries

### Dialect

databricks

### Version

2.3.4

### Configuration

```
[sqlfluff]
templater = jinja
dialect = databricks
sql_file_exts = .sql
max_line_length = 125
exclude_rules = ST06, AM05, L014, L029

[sqlfluff:templater:jinja]
apply_dbt_builtins = True
load_macros_from_path=macros

[sqlfluff:indentation]
tab_space_size = 4
indent_unit = space

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:layout:type:alias_expression]
# We want non-default spacing _before_ the alias expressions.
spacing_before = align
# We want to align them within the next outer select clause.
# This means for example that alias expressions within the FROM
# or JOIN clause would _not_ be aligned with them.
align_within = select_clause
# The point at which to stop searching outward for siblings, which
# in this example would likely be the boundary of a CTE. Stopping
# when we hit brackets is usually a good rule of thumb for this
# configuration.
align_scope = bracketed

[sqlfluff:rules:references.keywords]
# Comma separated list of words to ignore for this rule
ignore_words = element, no, rownum, resource, date, type, month, year, day, leading
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by chraltro** [comment]
+1 for this.

The issue is, as you can see if you run sqlfluff render, that the macro is parsed. So sqlfluff attempts to align the alias based on the parsed SQL, not the raw SQL.

That means, that if you try to align the alias using the Jinja templater, the macro line is replaced with the single macro name.

If you try to align the alias using the dbt templater, the macro line is replaced with the entire executed macro! That means that the aliases for the other columns can be pushed hundreds of characters to the right.

I think this is clearly a bug, and a very significant one for our team at least.

**Comment by vperron** [comment]
Can confirm. I have the same behaviour as soon as a macro or templating is used.

**Comment by MDTGRD** [comment]
We are also experiencing the same issues. Makes the fix command more cumbersome to use as you have to manually realign the CTE's with these cases.

**Comment by thangnv2212** [comment]
+1

**Comment by kyodjinn7** [comment]
same issue
